### PR TITLE
Update soccer to use newer common and remove debug communication code

### PR DESCRIPTION
**Pull request by jpfeltracco**
Software and firmware have diverged in their versions of common due to changes in constants by software, and a restructure and clean up in firmware.

This is making it hard to make changes to common that are needed in both software and firmware, specifically sending encoder values back to software (a motion control project) and adding the robot model to common.

This PR recitifes those differences. Note that firmware's version of common deleted both the debug communication header and the status header. Since debug communication is getting replaced by @petersonev by comp, I've deleted the debug communcation header and related code. I've also re-added the status.h header until we decide on what exactly will replace the rtp header (we may be moving back to protobuf).

### CODE REVIEW

**Comment by jpfeltracco** [comment]
@jgkamat Any idea what's happening with checkstyle?

**Comment by jgkamat** [comment]
Hmm, there seems to be new git integration in clang-format, making a binary called `git-clang-format`. However, stylize's guess on which clang format to use asks it to use the new file, which dosen't support the same options...

I'm pushing a new commit which will try to resolve this. I'm not sure why it's only happening on your branch though.

File: soccer/MainWindow.cpp [hunk 1]
```diff
@@ -625,8 +594,8 @@ void MainWindow::updateViews() {
             }
             statusWidget->setRobotModel(robotModel);
 
-// uncomment this #define to test the display of a variety of different errors
-// #define DEMO_ROBOT_STATUS
+            // uncomment this #define to test the display of a variety of
+            // different errors #define DEMO_ROBOT_STATUS
```
Comment (jgkamat) [inline soccer/MainWindow.cpp]: This is probably unintended (shakes fist at clang)

File: soccer/Robot.hpp [hunk 2]
```diff
@@ -4,30 +4,30 @@
...
+#include "planning/RotationCommand.hpp"
 
+#include <protobuf/Control.pb.h>
 #include <protobuf/RadioRx.pb.h>
 #include <protobuf/RadioTx.pb.h>
-#include <protobuf/Control.pb.h>
 #include <Utils.hpp>
 
+#include <stdint.h>
+#include <Eigen/Dense>
```
Comment (ashaw596) [inline soccer/Robot.hpp]: This stuff should probably be in the cpp if possible

File: soccer/Robot.hpp [hunk 3]
```diff
@@ -4,30 +4,30 @@
...
 #include "planning/DynamicObstacle.hpp"
+#include "planning/RotationCommand.hpp"
 
+#include <protobuf/Control.pb.h>
 #include <protobuf/RadioRx.pb.h>
 #include <protobuf/RadioTx.pb.h>
-#include <protobuf/Control.pb.h>
 #include <Utils.hpp>
 
+#include <stdint.h>
```
Comment (ashaw596) [inline soccer/Robot.hpp]: IS this suppose to be here?

**Comment by ashaw596** [comment]
Yeahhh, this stuff should defs be rewritten. THe packets were designed really hacky =P

File: soccer/Robot.hpp (hunk 2 above)
Comment (jpfeltracco) [reply soccer/Robot.hpp]: There's a function returning Eigen stuff so it should be in the header, same with stdint.h, there's a function using uint8_t

**Comment by jpfeltracco** [comment]
@petersonev has made good progress on a replacement, it'll just be a little longer before it's in. Is this branch okay to merge? The diff is a little nasty because of clang-format but the changes were pretty minimal.

**Comment by jpfeltracco** [comment]
Cool, merging.

**Pull request merged**